src/SemiclassicalOrthogonalPolynomials.jl

```diff
@@ -292,22 +292,33 @@ end
 semijacobi_ldiv(Q::SemiclassicalJacobi, P::SemiclassicalJacobi) = semijacobi_ldiv(Q.t, Q.a, Q.b, Q.c, P)
 function semijacobi_ldiv(Qt, Qa, Qb, Qc, P::SemiclassicalJacobi)
     @assert Qt ≈ P.t
-    (Qt ≈ P.t) && (Qa ≈ P.a) && (Qb ≈ P.b) && (Qc ≈ P.c) && return SymTridiagonal(Ones(∞),Zeros(∞))
-    Δa = Qa-P.a
-    Δb = Qb-P.b
-    Δc = Qc-P.c
+    (Qt ≈ P.t) && (Qa ≈ P.a) && (Qb ≈ P.b) && (Qc ≈ P.c) && return SymTridiagonal(Ones(∞), Zeros(∞))
+    Δa = Qa - P.a
+    Δb = Qb - P.b
+    Δc = Qc - P.c
     if isinteger(Δa) && isinteger(Δb) && isinteger(Δc) # (Δa,Δb,Δc) are integers -> use QR/Cholesky iteratively
-        if ((isone(Δa)||isone(Δa/2)) && iszero(Δb) && iszero(Δc)) || (iszero(Δa) && (isone(Δb)||isone(Δb/2)) && iszero(Δc))  || (iszero(Δa) && iszero(Δb) && (isone(Δc)||isone(Δc/2)))
+        if Δa < 0
+            R = ApplyArray(inv, semijacobi_ldiv(P.t, P.a, P.b, P.c, SemiclassicalJacobi(P.t, Qa, P.b, P.c)))
```

Comment:
The constructor `SemiclassicalJacobi(P.t, Qa, P.b, P.c)` should be avoided at all costs: we want to make sure we are building the new OP from the existing one as otherwise we have to redo an expensive Cholesky.

Reply:
Should they all just be `SemiclassicalJacobi(t, a, b, c, P)`?

Reply:
If `Qa < P.a` then `SemiclassicalJacobi(t, a, b, c, P)` won't work because lowering isn't implemented.

Reply:
Then shouldn't this be doing it the other way around and not calling `inv`?

Btw the extra function `semijacobi_ldiv(Qt, Qa, Qb, Qc, P::SemiclassicalJacobi)` is confusing. I think there should just be the one `semijacobi_ldiv(Q ::SemiclassicalJacobi, P ::SemiclassicalJacobi)`

Reply:
Actually scratch that. I think it should be `SemiclassicalJacobi(P.t, Qa, P.b, P.c, Q)`.

Reply:
The extra function `semijacobi_ldiv(Qt, Qa, Qb, Qc, P::SemiclassicalJacobi)` was really confusing me too actually. I'll use `semijacobi_ldiv(Q, P)` as you suggest.

> Then shouldn't this be doing it the other way around and not calling inv?

My thought was that, to go from $P^{t, (a, b, c)} \to P^{t, (a+\Delta t, b + \Delta b, c + \Delta c)}$, to go in steps. If for example $\Delta a < 0$, then the existing code lets us do $P^{t, (a + \Delta a, b, c)} \to P^{t, (a, b, c)}$ (raising) so that $P^{t, (a, b, c)} \to P^{t, (a + \Delta, b, c)}$ is just the inverse. This is the same as what's done in the `Jacobi` code from ClassicalOrthogonalPolynomials.jl pretty much.

Reply:
As long as a function like semijacobi_ldiv_direct(Qt, Qa, Qb, Qc, P::SemiclassicalJacobi) stays around the removal of semijacobi_ldiv(Qt, Qa, Qb, Qc, P::SemiclassicalJacobi) won't matter. If you don't like semijacobi_ldiv_direct then it will need a bigger rethink

Reply:
Keeping the `Q` around seems nice since it let's us reuse it in the constructors right? Rather than destructuring it into just the parameters

Reply:
Just in case there is confusion as to the convention: Conceptually, `semijacobi_ldiv_direct` goes the whole distance in one decomposition step, so for stability reasons should only be used for changes of +1 in a parameter but in principle it can be used for any positive polynomial modification. `semijacobi_ldiv` iterates over `semijacobi_ldiv_direct` to modify the paramters up in +1 (/ -1 once implemented) increments, which is stable.

Reply:
Would it make more sense then to just try and get lowering working first, then? I've not looked much into the details of it but maybe it makes things simpler.

Reply:
You should modify `semijacobi_ldiv_direct` with a case (-1,0,0) and so on, then call them in the same style it's currently done in `semijacobi_ldiv`, so that it iterates by calling itself recursively until all changes are accounted for.

To do the (-1,0,0) case it's totally fine to do the (1,0,0) case, i.e. the current code (note here the `M` are not optional)

```Julia
M = cholesky(P.X).U
return ApplyArray(*, Diagonal(Fill(1/M[1],∞)), M)
```
and then instead return the inv of that. I think that is what you have been working towards?

Reply:
If the inv plug and play there doesn't work (as an ApplyArray) then we can write a custom struct here to make it work. But once that mod in `semijacobi_ldiv_direct` is done it should all just work automatically. I could do a working prototype pretty quickly if you guys really need lowering

Reply:
Ah okay, yes I think that sounds like what I'm working towards. I didn't fully understand the `semijacobi_ldiv_direct` design - I should read through it and the details in the paper again. A working prototype would be pretty useful if you can, I'd appreciate that. We need lowering for my $b = -1$ work. I'd work at it now but I need to get my computer in for repairs sometime in the week before I can use Julia again.

> it would be nicer tho if inv of an UpperTriangular just worked there, perhaps we need to re-wrap with UpperTriangular, I don't know if the structure wrapper is maintained.

I've been looking at `inv` of an UpperTriangular while trying to get all this working that would make a lot of the issues I've been having here simpler. Need to have something like https://github.com/JuliaArrays/LazyArrays.jl/issues/320 resolved first (I think).